### PR TITLE
Opal 230 refactor kalendar controller

### PR DESCRIPTION
**Pull request by pataricza1987**

### CODE REVIEW

**Review by sznemeth3 — DISMISSED** [review]

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/KalendarController.java [hunk 1]
```diff
@@ -43,27 +43,24 @@
...
     @GetMapping(value = "/calendar")
-    public ResponseEntity getKalendarList(@RequestHeader("X-Client-Token") String clientToken, HttpServletRequest request) throws IOException {
-        if (!request.getHeader("X-Client-Token").equals("")) {
-            KalendarListResponse kalendarListResponse = new KalendarListResponse();
-            KalUser user = kalUserRepository.findByClientToken(clientToken);
-            List<Kalendar> list = kalendarService.findKalendars(user);
-            kalendarListResponse.setKalendars(kalendarService.setKalendarResponse(list));
+    public ResponseEntity getKalendarList(@RequestHeader("X-Client-Token") String clientToken) {
+        try {
+            KalendarListResponse kalendarListResponse = kalendarService.makeKalendarListResponse(clientToken);
```
Comment (arnikaeva) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/KalendarController.java]: I was thinking about renaming this method to be more descriptive - so when you check the KalendarService class it is meaningful and you do not need to check this calles method to understand what is happening.
Maybe `kalendarService.getKalendarsByUser` or `kalendarService.getKalendarsByClientToken` ?

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java [hunk 2]
```diff
@@ -43,11 +54,54 @@
...
+        KalendarListResponse kalendarListResponse = new KalendarListResponse();
+        List<KalendarResponse> kalendarResponse = setKalendarResponse(clientToken);
+        kalendarListResponse.setKalendars(kalendarResponse);
+        return kalendarListResponse;
+    }
+
+    public List<KalendarResponse> setKalendarResponse(String clientToken) throws ValidationException {
+        List<Kalendar> kalendars = findKalendars(clientToken);
+        List<KalendarResponse> kalendarResponses = new ArrayList<>();
+        for (int i = 0; i < kalendars.size(); i++) {
```
Comment (arnikaeva) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java]: use streaming+ builder pattern for creating kalendarResponse

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java [hunk 3]
```diff
@@ -43,11 +54,54 @@
...
+            KalendarResponse kalendarResponse = new KalendarResponse();
+            kalendarResponse.setOutputGoogleAuthId(kalendars.get(i).getOutputGoogleAuthId());
+            kalendarResponse.setOutputCalendarId(kalendars.get(i).getName());
+            kalendarResponse.setId(kalendars.get(i).getId());
+            kalendarResponse.setInputGoogleCalendars((setToStringGoogleCalendars(findGoogleCalendarsByKalendar(kalendars.get(i)))));
```
Comment (arnikaeva) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java]: nested method calls are hard to maintain and debug

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java [hunk 4]
```diff
@@ -25,6 +26,16 @@
 @Service
 public class KalendarService {
 
+    private static final String USER_NOT_FOUND_TOKEN = "User not found for clientToken=";
```
Comment (arnikaeva) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java]: I would not put exception messages as constants if we are not reusing them anywhere else

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java (hunk 2 above)
Comment (arnikaeva) [reply src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/KalendarService.java]:
```java
List<Kalendar> kalendars = findKalendars(clientToken);
List<KalendarResponse> kalendarResponses = kalendars.stream
  .map(kalendar -> {
    new KalendarResponseBuilder
      .setOutputGoogleAuthId(kalendar.getOutputGoogleAuthId())
      .setId(..)
      ....
      .build();
   })
   .collectors(Collector.toList());
```

**Review by arnikaeva — CHANGES_REQUESTED** [review]
Refactoring :)

File: src/main/java/com/greenfoxacademy/kalendaryo/model/builder/KalendarResponseBuilder.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    return this;
+  }
+
+  public KalendarResponseBuilder setId(long id) {
+    this.id = id;
+    return this;
+  }
+
+  public KalendarResponse build() {
+    return  new KalendarResponse(outputGoogleAuthId, outputCalendarId, inputGoogleCalendars, id);
```
Comment (arnikaeva) [inline src/main/java/com/greenfoxacademy/kalendaryo/model/builder/KalendarResponseBuilder.java]: unneccesary space after `return` 😄 
btw great work you could do a small lightning talk about the builder pattern sometime :)

**Pull request closed without merging**